drivers: sensors: nxp: kinetis: temp: wrong length provided to memset()

**Describe the bug**
The NXP Kinetis temperature sensor driver (`drivers/sensor/nxp/nxp_kinetis_temp/temp_kinetis.c`) uses `memset()` for zeroing the buffer used for storing ADC samples.

Problem is, the length of the buffer provided is calculated using `ARRAY_SIZE()`, whereas it should have been a simple `sizeof()`. This is causing a compiler warning (show below).

**To Reproduce**
Steps to reproduce the behavior:
1. `west build -b frdm_k64f/mk64f12 samples/hello_world/ -- -DCONFIG_SENSOR=y -DCONFIG_ADC=y`
2. See error

```
[ 82%] Building C object zephyr/drivers/sensor/nxp/nxp_kinetis_temp/CMakeFiles/drivers__sensor__nxp__nxp_kinetis_temp.dir/temp_kinetis.c.obj
/home/brix/Projects/zephyrproject/zephyr/drivers/sensor/nxp/nxp_kinetis_temp/temp_kinetis.c: In function 'temp_kinetis_init':
/home/brix/Projects/zephyrproject/zephyr/drivers/sensor/nxp/nxp_kinetis_temp/temp_kinetis.c:162:9: warning: 'memset' used with length equal to number of elements without multiplication by element size [-Wmemset-elt-size]
  162 |         memset(&data->buffer, 0, ARRAY_SIZE(data->buffer));
      |         ^~~~~~
[ 82%] Linking C static library libdrivers__sensor__nxp__nxp_kinetis_temp.a
```

**Expected behavior**
No compiler warning, proper zeroing of the ADC sample buffer.

**Impact**
Compiler warnings causing CI failure.

**Environment (please complete the following information):**
 - OS: Linux
 - Toolchain Zephyr SDK 0.16.5-1
 - Commit SHA: ec0e2d6308680203bb58275062b79105016ea1b1

**Additional context**
Add any other context that could be relevant to your issue, such as pin setting,
target configuration, ...

--- a/drivers/sensor/nxp/nxp_kinetis_temp/Kconfig
+++ b/drivers/sensor/nxp/nxp_kinetis_temp/Kconfig
@@ -7,7 +7,8 @@ config TEMP_KINETIS
 	bool "NXP Kinetis Temperature Sensor"
 	default y
 	depends on DT_HAS_NXP_KINETIS_TEMPERATURE_ENABLED
-	depends on (ADC && SOC_FAMILY_KINETIS)
+	depends on SOC_FAMILY_KINETIS
+	select ADC
 	help
 	  Enable driver for NXP Kinetis temperature sensor.
 

--- a/drivers/sensor/nxp/nxp_kinetis_temp/temp_kinetis.c
+++ b/drivers/sensor/nxp/nxp_kinetis_temp/temp_kinetis.c
@@ -159,7 +159,7 @@ static int temp_kinetis_init(const struct device *dev)
 		},
 	};
 
-	memset(&data->buffer, 0, ARRAY_SIZE(data->buffer));
+	memset(&data->buffer, 0, sizeof(data->buffer));
 
 	if (!device_is_ready(config->adc)) {
 		LOG_ERR("ADC device is not ready");

